src/vec/tests.rs

```diff
@@ -229,6 +229,27 @@ fn test_iter() {
     assert_eq!(sv.iter().skip(usize::MAX).count(), 0);
 }
 
+#[test]
+fn test_iter_count() {
+    let sv = StableVec::<u64, M>::new(M::default()).unwrap();
+    sv.push(&1).unwrap();
+    sv.push(&2).unwrap();
+    sv.push(&3).unwrap();
+    sv.push(&4).unwrap();
+    {
+        let mut iter = sv.iter();
+        iter.next_back();
+        assert_eq!(iter.count(), 3);
```

Comment:
Before this PR, this returns 4. The stdlib `Vec` returns 3 so we should also return 3.